File: factory-method/c#/Animals/Chicken.cs
namespace FactoryMethod
{
    public class Chicken : IAnimal
    {
        public decimal Weight { get; set; }

        public ICookedMeat Cook()
        {
            return Grill();
        }

        private GrilledChicken Grill()
        {
            var theMeat = new GrilledChicken();
            theMeat.Weight = Weight * 0.8M; //'M' is just used here to define a Decimal literal
            return theMeat;
        }
    }
}

File: factory-method/typescript/src/Animals/Chicken.ts
import {IAnimal} from "../Interfaces/IAnimal";
import {ICookedMeat} from "../Interfaces/ICookedMeat";
import {GrilledChicken} from "../Meats/GrilledChicken";
export class Chicken implements IAnimal{
  public Weight:number;
  public Cook() : ICookedMeat{
    return this.Grill();
  }
  private Grill() : GrilledChicken{
    let theMeat : GrilledChicken = new GrilledChicken();
    theMeat.Weight = this.Weight * 0.8;
    return theMeat;
  }
}